internal/uuid/uuid.go

```diff
@@ -0,0 +1,229 @@
+)
+
+// Constants defining our field widths.
+const (
+	// customEpoch is January 1, 2025 in microseconds since the Unix epoch.
+	customEpoch uint64 = 1735689600000000
+
+	timestampBits = 52 // Timestamp: microseconds since customEpoch.
+	sequenceBits  = 8  // Sequence counter: allows 256 per microsecond.
+	versionBits   = 4  // Version field.
+	typeBits      = 8  // Type identifier.
+	unusedBits    = 56 // Unused bits.
+)
+
+var (
+	// NilUUID is the zero-valued UUID.
+	NilUUID = UUID{}
+	// Generator singleton.
+	generator = newUUIDGenerator()
+	// currentVersion is the default version.
+	currentVersion uint8 = 1
+)
+
+type IDType uint8
+
+// UUID is a 128-bit identifier.
+type UUID [16]byte
+
+// Bit layout (big-endian bits 127..0):
+//  • Bits 127–76: Timestamp (52 bits)
+//  • Bits 75–68: Sequence (8 bits)
+//  • Bits 67–64: Version (4 bits)
+//  • Bits 63–60: Type high nibble (4 bits)
+//  • Bits 59–56: Type low nibble (4 bits)
+//  • Bits 55–0 : Unused (56 bits)
+
+// Time returns the time the UUID was generated.
+func (u UUID) Time() time.Time {
+	t := int64(u.Timestamp() + customEpoch)
+	seconds := t / 1e6
+	nanoseconds := (t % 1e6) * 1000
+	return time.Unix(seconds, nanoseconds)
+}
+
+// Timestamp returns the 52-bit timestamp (microseconds since Jan 1, 2025).
+func (u UUID) Timestamp() uint64 {
+	high := binary.BigEndian.Uint64(u[0:8])
+	return high >> (sequenceBits + versionBits + (typeBits / 2))
+}
+
+// Sequence returns the 8-bit sequence counter.
+func (u UUID) Sequence() uint8 {
+	high := binary.BigEndian.Uint64(u[0:8])
+	return uint8((high >> (versionBits + (typeBits / 2))) & ((1 << sequenceBits) - 1))
+}
+
+// Version returns the 4-bit version.
+func (u UUID) Version() uint8 {
+	high := binary.BigEndian.Uint64(u[0:8])
+	return uint8((high >> (typeBits / 2)) & ((1 << versionBits) - 1))
+}
+
+// Type returns the 8-bit type identifier.
+func (u UUID) Type() IDType {
+	high := binary.BigEndian.Uint64(u[0:8])
+	low := binary.BigEndian.Uint64(u[8:16])
+	typeHigh := high & ((1 << (typeBits / 2)) - 1)
+	typeLow := (low >> unusedBits) & ((1 << (typeBits / 2)) - 1)
+	return IDType((typeHigh << (typeBits / 2)) | typeLow)
+}
+
+// Unused returns the 56-bit unused field.
+func (u UUID) Unused() uint64 {
+	low := binary.BigEndian.Uint64(u[8:16])
+	return low & ((uint64(1) << unusedBits) - 1)
+}
+
+// IsNil indicates whether the UUID is the zero value.
+func (u UUID) IsNil() bool {
+	return u == NilUUID
+}
+
+// String returns the UUID as a formatted string.
+// "<timestamp:13 hex digits><sequence:2 hex digits>-<version:1 hex digit>-<type:2 hex digits>-<unused:14 hex digits>"
+func (u UUID) String() string {
+	ts := u.Timestamp()
+	seq := u.Sequence()
+	ver := u.Version()
+	typ := u.Type()
+	un := u.Unused()
+	return fmt.Sprintf("%013x%02x-%01x-%02x-%014x", ts, seq, ver, typ, un)
+}
+
+// FromString parses a UUID from its string representation.
+func FromString(s string) (UUID, error) {
+	parts := strings.Split(s, "-")
+	if len(parts) != 4 {
```

Comment:
This is a strict split by `-`, is it possible to check only by length and version format definition? I mean use - split only as a display format not as a mandatory format.

Reply:
Sure i think we can do that... where would the other format come from if String() only returns 1 style?

Reply:
In the future, if there are multiple versions of UUIDs, they may not be displayed in the same format, so the only thing we need to make sure is to get the version number and parse it based on the version format, without presupposing how many `-` and how long each segment will be.

Of course if it is guaranteed that this format will not change then there is no need to consider this issue.

Reply:
Ah I see what you mean.

Yeah, I made a note to move version to the front position so we can change the rules as needed.

Reply:
Moved the version to the front.
"nil" UUID's are now any UUID that starts with `0` (zero) or empty strings (`""`).
String()'ified nil UUID's are simply `"0"` now.

`FromString()` Checks for version before deciding on a specific decoding function to use. Only version 1 supported at the moment (`fromString_v1()`), Any other version fails over to a nil return value.

`String()` Checks for version before deciding on a specific encoding function to use. Only version 1 supported at the moment (`toString_v1()`).

In theory in old version UUID that is decoded, will only re-encode into the current version format. That's the goal anyways.

Reply:
LGTM